spec/tree_spec.rb

```diff
@@ -1,14 +1,240 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+describe Tree do
+  let(:tree) { Tree.new }
+
+  it 'should be an AppleTree' do
+    expect(tree.is_a? Tree).to be true
+  end
+
+  it 'should have a height of 0' do
+    expect(tree.height).to be 0  
+  end
+
+  it 'should have an age of 0' do
+    expect(tree.age).to be 0
+  end
+
+  it 'should have an empty set of apples' do
+    expect(tree.apples).to eq []
+  end
+
+  describe '#age!' do
+    
+    context 'after aging once' do
+      before(:each) { tree.age! }
+
+      it 'age is 1' do
+        expect(tree.age).to be 1
+      end
+
+      it 'height is 1' do
+        expect(tree.height).to be 1
+      end
+
+      it 'has 5 apples' do
+        expect(tree.apples.length).to be 5     
+      end
```

Comment:
It's important that tests fail when something in the code is changed, because the developer who changed it should know that functionality is changing. But I'm glad you understand that this test about aging doesn't really have enough to do with apples for it to be affected if add_apples changes. As long as the add_apples test fails, it's good. How about you try mocking it out?